templates/vue/src/components/NodeModal.vue

```diff
@@ -334,6 +334,106 @@
+              <file-upload
+                v-model="node.typeData.spaceshipPartNotEarnedIconUrl"
+                placeholder="Enter link (starting with http)"
+              />
+            </b-form-group>
+            <b-form-group label="Spaceship Part - earned">
+              <file-upload
+                v-model="node.typeData.spaceShipPartEarnedIconUrl"
+                placeholder="Enter link (starting with http)"
+              />
+            </b-form-group>
+            <b-form-group label="Spaceship Part - hover">
+              <file-upload
+                v-model="node.typeData.spaceShipPartHoverUrl"
+                placeholder="Enter link (starting with http)"
+              />
+            </b-form-group>
+            <label for="node-spaceship-parts">
+              Spaceship Part square coordinates
+            </label>
+            <b-row id="node-spaceship-parts">
+              <b-col
+                sm="3"
+                id="node-shapeship-coordinate-text"
+              >
+                Upper lefthand corner
+              </b-col>
+              <b-col>
+                <b-form-group
+                  label-cols="4"
+                  label-cols-lg="2"
+                  label="X: "
+                  label-for="node-spaceship-part-x"
+                >
+                  <b-form-input
+                    id="node-spaceship-part-x"
+                    v-model="node.typeData.spaceshipPartX"
+                    placeholder="In pixels (top left)"
+                  />
+                </b-form-group>
+              </b-col>
+              <b-col>
+                <b-form-group
+                  label-cols="4"
+                  label-cols-lg="2"
+                  label="Y: "
+                  label-for="node-spaceship-part-y"
+                >
+                  <b-form-input
+                    id="node-spaceship-part-y"
+                    v-model="node.typeData.spaceshipPartY"
+                    placeholder="In pixels (top left)"
+                  />
+                </b-form-group>
+              </b-col>
+            </b-row>
+            <b-form-group
+              label-cols="4"
+              label-cols-lg="2"
+              label="Width: "
+              label-for="node-spaceship-part-width"
+            >
+              <b-form-input
+                id="node-spaceship-part-width"
+                v-model="node.typeData.spaceshipPartWidth"
+                placeholder="In pixels"
+              />
+            </b-form-group>
+            <b-form-group
+              label-cols="4"
+              label-cols-lg="2"
+              label="Height: "
+              label-for="node-spaceship-part-height"
+            >
+              <b-form-input
+                id="node-spaceship-part-height"
+                v-model="node.typeData.spaceshipPartHeight"
+                placeholder="In pixels"
+              />
+            </b-form-group>
```

Comment:
It would be nice if we organize this the same way as the coordinates so they are both in the same row (called "Dimensions" or something)